### PR TITLE
Make the calendar's date formatter translatable

### PR DESCRIPTION
**Pull request by gagnieray**
Related to #256

### CODE REVIEW

File: galette/templates/default/elements/js/calendar.js.twig
Comment (trasher) [inline galette/templates/default/elements/js/calendar.js.twig:5]: You can either use `__()` or `_T('string', 'domain', false)` ;)

File: galette/lib/Galette/Features/Replacements.php
Comment (trasher) [inline galette/lib/Galette/Features/Replacements.php:229]: Comments will be displayed in weblate UI
image

**Review by trasher — APPROVED** [review]
Looks good to me; @gagnieray if you can test if I did not break anything... thanks ;)

**Comment by gagnieray** [comment]
> Looks good to me; @gagnieray if you can test if I did not break anything... thanks ;)

After a quick test everything seems to work fine in languages other than french.
Shouldn't this work even in french with `fui_dateformatter` not yet translated ?

Member creation/modification is OK

But it's not the case for contributions : 

```
Type : ErrorException

Code : 0

Message : DateTime::__construct(): Passing null to parameter #1 ($datetime) of type string is deprecated

Fichier : /home/www/galette/galette/lib/Galette/Entity/Contribution.php

Ligne : 251 
```

```
Trace

#0 [internal function]: {closure}()
#1 /home/www/galette/galette/lib/Galette/Entity/Contribution.php(251): DateTime->__construct()
#2 /home/www/galette/galette/lib/Galette/Entity/Contribution.php(507): Galette\Entity\Contribution->retrieveEndDate()
#3 /home/www/galette/galette/lib/Galette/Controllers/Crud/ContributionsController.php(820): Galette\Entity\Contribution->check()
#4 /home/www/galette/galette/lib/Galette/Controllers/Crud/ContributionsController.php(224): Galette\Controllers\Crud\ContributionsController->store()
#5 [internal function]: Galette\Controllers\Crud\ContributionsController->doAdd()
#6 /home/www/galette/galette/vendor/php-di/invoker/src/Invoker.php(74): call_user_func_array()
#7 /home/www/galette/galette/vendor/php-di/slim-bridge/src/ControllerInvoker.php(47): Invoker\Invoker->call()
#8 /home/www/galette/galette/vendor/slim/slim/Slim/Routing/Route.php(358): DI\Bridge\Slim\ControllerInvoker->__invoke()
#9 /home/www/galette/galette/lib/Galette/Middleware/Authenticate.php(201): Slim\Routing\Route->handle()
#10 /home/www/galette/galette/vendor/slim/slim/Slim/MiddlewareDispatcher.php(269): Galette\Middleware\Authenticate->__invoke()
#11 /home/www/galette/galette/vendor/slim/slim/Slim/MiddlewareDispatcher.php(65): Psr\Http\Server\RequestHandlerInterface@anonymous->handle()
#12 /home/www/galette/galette/vendor/slim/slim/Slim/MiddlewareDispatcher.php(65): Slim\MiddlewareDispatcher->handle()
#13 /home/www/galette/galette/vendor/slim/slim/Slim/Routing/Route.php(315): Slim\MiddlewareDispatcher->handle()
#14 /home/www/galette/galette/vendor/slim/slim/Slim/Routing/RouteRunner.php(68): Slim\Routing\Route->run()
#15 /home/www/galette/galette/vendor/akrabat/rka-slim-session-middleware/RKA/SessionMiddleware.php(66): Slim\Routing\RouteRunner->handle()
#16 /home/www/galette/galette/vendor/slim/slim/Slim/MiddlewareDispatcher.php(121): RKA\SessionMiddleware->process()
#17 /home/www/galette/galette/vendor/slim/csrf/src/Guard.php(476): Psr\Http\Server\RequestHandlerInterface@anonymous->handle()
#18 /home/www/galette/galette/vendor/slim/slim/Slim/MiddlewareDispatcher.php(121): Slim\Csrf\Guard->process()
#19 /home/www/galette/galette/lib/Galette/Middleware/TrailingSlash.php(89): Psr\Http\Server\RequestHandlerInterface@anonymous->handle()
#20 /home/www/galette/galette/vendor/slim/slim/Slim/MiddlewareDispatcher.php(168): Galette\Middleware\TrailingSlash->__invoke()
#21 /home/www/galette/galette/lib/Galette/Middleware/Language.php(86): Psr\Http\Server\RequestHandlerInterface@anonymous->handle()
#22 /home/www/galette/galette/vendor/slim/slim/Slim/MiddlewareDispatcher.php(168): Galette\Middleware\Language->__invoke()
#23 /home/www/galette/galette/lib/Galette/Middleware/Telemetry.php(86): Psr\Http\Server\RequestHandlerInterface@anonymous->handle()
#24 /home/www/galette/galette/vendor/slim/slim/Slim/MiddlewareDispatcher.php(168): Galette\Middleware\Telemetry->__invoke()
#25 /home/www/galette/galette/includes/main.inc.php(206): Psr\Http\Server\RequestHandlerInterface@anonymous->handle()
#26 /home/www/galette/galette/vendor/slim/slim/Slim/MiddlewareDispatcher.php(269): Closure->{closure}()
#27 /home/www/galette/galette/vendor/slim/slim/Slim/Middleware/RoutingMiddleware.php(45): Psr\Http\Server\RequestHandlerInterface@anonymous->handle()
#28 /home/www/galette/galette/vendor/slim/slim/Slim/MiddlewareDispatcher.php(121): Slim\Middleware\RoutingMiddleware->process()
#29 /home/www/galette/galette/vendor/slim/slim/Slim/Middleware/ErrorMiddleware.php(76): Psr\Http\Server\RequestHandlerInterface@anonymous->handle()
#30 /home/www/galette/galette/vendor/slim/slim/Slim/MiddlewareDispatcher.php(121): Slim\Middleware\ErrorMiddleware->process()
#31 /home/www/galette/galette/vendor/slim/twig-view/src/TwigMiddleware.php(115): Psr\Http\Server\RequestHandlerInterface@anonymous->handle()
#32 /home/www/galette/galette/vendor/slim/slim/Slim/MiddlewareDispatcher.php(121): Slim\Views\TwigMiddleware->process()
#33 /home/www/galette/galette/vendor/slim/slim/Slim/MiddlewareDispatcher.php(65): Psr\Http\Server\RequestHandlerInterface@anonymous->handle()
#34 /home/www/galette/galette/vendor/slim/slim/Slim/App.php(199): Slim\MiddlewareDispatcher->handle()
#35 /home/www/galette/galette/vendor/slim/slim/Slim/App.php(183): Slim\App->handle()
#36 /home/www/galette/galette/includes/main.inc.php(238): Slim\App->run()
#37 /home/www/galette/galette/webroot/index.php(57): require_once('...')
#38 {main}
```

On contribution creation there is also something missing : on member selection, the ajax call reformat the membership starting date to DD/MM/YYYY, and then of course the validation error message is displayed :

Capture d’écran de 2023-02-28 20-06-33

Same when using filters : 

Capture d’écran de 2023-02-28 19-56-11

File: galette/lib/Galette/Features/Replacements.php
Comment (gagnieray) [reply galette/lib/Galette/Features/Replacements.php:229]: I really should have a look at weblate one day ! :sweat_smile:

**Comment by trasher** [comment]
> Member creation/modification is OK
> 
> But it's not the case for contributions :

Well.. Contributions are not working, but that sounds legit, since the new string is not translated; therefore date formats are differents.
Honestly, I was a bit surprised it worked for members :D

Well, I'll take a deeper look, and maybe try to add tests on that, but I think members setter method may do a bit of magic with a fallback on US date format (db one) that is not done on contributions.

**Comment by trasher** [comment]
Last commit should fix issue for Contributions and Transactions

**Comment by trasher** [comment]
I'll merge this one; even if there are still issues; I'll fix them on other PRs

**Comment by trasher** [comment]
Manually merged

**Pull request closed without merging**
